MapboxNavigationUI/RouteMapViewController.swift

```diff
@@ -240,7 +240,46 @@ extension RouteMapViewController: NavigationMapViewDelegate {
             }
         }
         
-        return CLLocation(coordinate: newCoordinate, altitude: location.altitude, horizontalAccuracy: location.horizontalAccuracy, verticalAccuracy: location.verticalAccuracy, course: location.course, speed: location.speed, timestamp: location.timestamp)
+        let defaultReturn = CLLocation(coordinate: newCoordinate, altitude: location.altitude, horizontalAccuracy: location.horizontalAccuracy, verticalAccuracy: location.verticalAccuracy, course: location.course, speed: location.speed, timestamp: location.timestamp)
+        
+        guard location.course == -1 else {
+            return defaultReturn
+        }
+        
+        guard let coords = routeController.routeProgress.route.coordinates else {
```

Comment:
`coordinates` is only `nil` if `RouteOptions.routeShapeResolution` was set to `.none`, in which case we’ve got much bigger problems. Force-unwrap (`!`) `coordinates` instead of silently failing here. We should also add an assertion as soon as possible that `coordinates` is non-nil, probably in `RouteController(route:)`.